plugins/integration/providers/voicebase/lib/KVoicebaseIntegrationEngine.class.php

```diff
@@ -69,6 +70,16 @@ protected function doDispatch(KalturaBatchJob $job, KalturaIntegrationJobData &$
 	
 	protected function doClose(KalturaBatchJob $job, KalturaIntegrationJobData &$data, KalturaVoicebaseJobProviderData $providerData)
 	{
+		$entryId = $providerData->entryId;
+		$this->clientHelper = VoicebasePlugin::getClientHelper($providerData->apiKey, $providerData->apiPassword);
+		$remoteProcess = $this->clientHelper->retrieveRemoteProcess($entryId);
+		
+		//false result means that something has gone wrong - the VB job is either in status error or missing altogether
+		if(!$remoteProcess || $remoteProcess->requestStatus == VoicebaseClientHelper::VOICEBASE_FAILURE_MEESAGE || !isset($remoteProcess->fileStatus) || $curlResult->fileStatus == VoicebaseClientHelper::VOICE_MACHINE_FAILURE_MESSAGE)
+		{
+			throw new Exception("VoiceBase transcription failed.");
+		}
+		
 		return false;
```

Comment:
We should also support returning true in case processing finished (for cases where notification is not received from VoiceBase)